HTML5 build error related to PCRE2

Compiling Godot Git https://github.com/godotengine/godot/commit/a717083235b3369aef9a3846b5071fac023a60b2 fails for the HTML5 (WebAssembly) platform, on an Ubuntu 16.04 host running the latest stable Emscripten version:

```
In file included from thirdparty/pcre2/src/pcre2_jit_compile.c:78:
thirdparty/pcre2/src/sljit/sljitLir.c:1754:49: error: conflicting types for 'sljit_create_compiler'
SLJIT_API_FUNC_ATTRIBUTE struct sljit_compiler* sljit_create_compiler(void)
                                                ^
thirdparty/pcre2/src/sljit/sljitLir.h:424:49: note: previous declaration is here
SLJIT_API_FUNC_ATTRIBUTE struct sljit_compiler* sljit_create_compiler(void *allocator_data);
                                                ^
thirdparty/pcre2/src/pcre2_jit_compile.c:81:2: error: Unsupported architecture
#error Unsupported architecture
 ^
Compiling ==> thirdparty/pcre2/src/pcre2_match.c
In file included from thirdparty/pcre2/src/pcre2_jit_compile.c:11499:
thirdparty/pcre2/src/pcre2_jit_misc.c:113:1: error: implicit declaration of function 'sljit_free_unused_memory_exec' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
sljit_free_unused_memory_exec();
^
3 errors generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
scons: *** [thirdparty/pcre2/src/pcre2_jit_compile_16.javascript.opt.debug.webassembly.bc] Error 1
scons: building terminated because of errors.
```

According to the error message, this is likely a regression from https://github.com/godotengine/godot/pull/10148.

--- a/modules/regex/SCsub
+++ b/modules/regex/SCsub
@@ -8,8 +8,11 @@ env_regex.Append(CPPFLAGS=["-DPCRE2_CODE_UNIT_WIDTH=0"])
 env_regex.add_source_files(env.modules_sources, "*.cpp")
 
 if (env['builtin_pcre2'] != 'no'):
-    thirdparty_dir = "#thirdparty/pcre2/src/"
-    thirdparty_flags = ["-DPCRE2_STATIC", "-DHAVE_CONFIG_H", "-DSUPPORT_JIT"]
+    jit_blacklist = ['javascript']
+    thirdparty_dir = '#thirdparty/pcre2/src/'
+    thirdparty_flags = ['-DPCRE2_STATIC', '-DHAVE_CONFIG_H']
+    if 'platform' in env and env['platform'] not in jit_blacklist:
+        thirdparty_flags.append('-DSUPPORT_JIT')
     thirdparty_sources = [
         "pcre2_auto_possess.c",
         "pcre2_chartables.c",

